ui/packages/plot/src/Plot.svelte

```diff
@@ -1,20 +1,55 @@
-<script>
+<script lang='ts'>
 	//@ts-nocheck
 	import Plotly from "plotly.js-dist-min";
 	import { Plot as PlotIcon } from "@gradio/icons";
+	import { colors as color_palette, ordered_colors } from "@gradio/theme";
+	import { get_next_color } from "@gradio/utils";
 	import { Vega } from "svelte-vega";
-
 	import { afterUpdate, onDestroy } from "svelte";
+	import {create_config} from "./utils"
 
 	export let value;
 	export let target;
 	let spec = null;
+	export let colors: Array<string> = [];
+	export let theme: string;
+	export let caption: string;
+	
+	function get_color(index: number) {
+		let current_color = colors[index % colors.length];
+
+		if (current_color && current_color in color_palette) {
+			return color_palette[current_color as keyof typeof color_palette]
+				?.primary;
+		} else if (!current_color) {
+			return color_palette[get_next_color(index) as keyof typeof color_palette]
+				.primary;
+		} else {
+			return current_color;
+		}
+	}
+
+	$: darkmode = theme == "dark"
 
 	$: if(value && value['type'] == "altair") {
 		spec = JSON.parse(value['plot'])
+		const config = create_config(darkmode);
+		spec['config'] = config;
```

Comment:
```suggestion
		spec.config = config;
```

This is more idiomatic.